curriculum/challenges/english/blocks/lab-debug-donation-form/690ddb5e6ee94eb73ed172b0.md

```diff
@@ -0,0 +1,180 @@
+---
+id: 690ddb5e6ee94eb73ed172b0
+title: Debug a Donation Form
+challengeType: 25
+dashedName: debug-donation-form
+# Purposefully removed the demo type because this is a debugging project.
+---
+
+# --description--
+
+A local charity has built a donation form website, but there are several issues that need to be fixed. The form isn't accessible and has some HTML syntax errors.
+
+Your job is to fix all of the errors so the form works correctly and is accessible to all users. Complete the items in the user stories below and click "Run the Tests" to see if you fixed all the errors.
+
+**User Stories:**
+
+1. The `input` elements are void elements and should not have closing tags. Remove all `</input>` closing tags from the form.
+2. Add `label` elements for each form input field so users know what each field is for. The label text should match what's currently next to each input.
+3. The `Email Address:` input type should be an `email` instead of `text`.
+4. You should associate each `label` element with its corresponding `input` element using the `for` attribute on the `label` and a matching `id` attribute on the `input`.
+5. Add the `required` attribute to the text, email, and number input fields (but not the checkbox or submit button) to ensure users fill in the required information.
+
+# --hints--
+
+You should not have any `</input>` closing tags in your code.
+
+```js
+assert.notMatch(code, /<\/input>/);
```

Comment:
It may be possible that instead of deleting, the closing input tags are change into elements, maybe you should check that there is the correct number of input elements?

Reply:
Something like this?

````md
```js
assert.lengthOf(document.querySelectorAll("input"), 5);
```
````